author	Syam Sidhardhan <s.syam@samsung.com>
commit	7730ff4a3615d4dd6961d5296ee7aab06ce0d5cb
	Mon, 17 Sep 2012 13:49:35 +0000 (16:49 +0300)

gstavdtpsink: Free the D-Bus error on reply failure

If D-Bus error is set we should free it.

diff -cr bluez-4.99.pan.orig/audio/gstavdtpsink.c bluez-4.99.pan/audio/gstavdtpsink.c
*** bluez-4.99.pan.orig/audio/gstavdtpsink.c	Tue Oct 16 15:46:41 2012
--- bluez-4.99.pan/audio/gstavdtpsink.c	Wed Oct 17 05:26:31 2012
***************
*** 1390,1402 ****
  	msg = dbus_message_new_method_call("org.bluez", self->transport,
  						"org.bluez.MediaTransport",
  						"GetProperties");
  	reply = dbus_connection_send_with_reply_and_block(self->data->conn,
  							msg, -1, &err);
  
! 	if (dbus_error_is_set(&err) || reply == NULL) {
! 		GST_ERROR_OBJECT(self, "Failed to get transport properties: %s",
! 					err.message);
! 		goto fail;
  	}
  
  	if (!dbus_message_iter_init(reply, &arg_i)) {
--- 1390,1408 ----
  	msg = dbus_message_new_method_call("org.bluez", self->transport,
  						"org.bluez.MediaTransport",
  						"GetProperties");
+ 	if (!msg) {
+ 		GST_ERROR_OBJECT(self, "D-Bus Memory allocation failed");
+ 		return FALSE;
+ 	}
+ 
  	reply = dbus_connection_send_with_reply_and_block(self->data->conn,
  							msg, -1, &err);
+ 	dbus_message_unref(msg);
  
! 	if (dbus_error_is_set(&err)) {
! 		GST_ERROR_OBJECT(self, "GetProperties failed: %s", err.message);
! 		dbus_error_free(&err);
! 		return FALSE;
  	}
  
  	if (!dbus_message_iter_init(reply, &arg_i)) {
***************
*** 1428,1434 ****
  	return gst_avdtp_sink_update_caps(self);
  
  fail:
- 	dbus_message_unref(msg);
  	dbus_message_unref(reply);
  	return FALSE;
  
--- 1434,1439 ----
